File: mns-core/src/main/java/cz/fb/manaus/core/manager/filter/MarketFilter.java
package cz.fb.manaus.core.manager.filter;

import cz.fb.manaus.core.model.Market;

import java.util.Set;

public interface MarketFilter {

    boolean accept(Market market, Set<String> categoryBlacklist);

    default boolean isStrict() {
        return false;
    }
}

File: mns-core/src/main/java/cz/fb/manaus/core/manager/filter/MarketFilter.kt
package cz.fb.manaus.core.manager.filter

import cz.fb.manaus.core.model.Market
import cz.fb.manaus.core.provider.ProviderSelector

interface MarketFilter : ProviderSelector {

    val isStrict: Boolean
        get() = false

    fun accept(market: Market): Boolean
}
